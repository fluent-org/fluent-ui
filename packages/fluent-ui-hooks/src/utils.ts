export const global = typeof window !== 'undefined' && window
