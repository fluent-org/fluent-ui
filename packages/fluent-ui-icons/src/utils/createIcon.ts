import * as React from 'react'
import styled from 'styled-components'

export type Children = [string, any]
export type JSX = ['svg', any, Children]

function createIcon(jsx: JSX, componentName: string): React.ForwardRefExoticComponent<{}> {
  const [type, props, children] = jsx
  function Icon(_props: any): any {
    return React.createElement(type, { ...props, ..._props }, React.createElement(...children))
  }
  const SvgIconStyled = styled(Icon)`
    width: 0.83em;
    height: 0.83em;
    display: inline-block;
    user-select: none;
    font-size: inherit;
    color: inherit;
    fill: currentColor;
  `
  SvgIconStyled.displayName = componentName
  return SvgIconStyled
}

export default createIcon
