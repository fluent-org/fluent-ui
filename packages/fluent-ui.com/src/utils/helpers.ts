export function getCookie(name: string): string {
  const regex = new RegExp(`(?:(?:^|.*;*)${name}*=*([^;]*).*$)|^.*$`)
  return document.cookie.replace(regex, '$1')
}
