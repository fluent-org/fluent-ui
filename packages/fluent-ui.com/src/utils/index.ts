export * from './scroll'
export * from './theme'
