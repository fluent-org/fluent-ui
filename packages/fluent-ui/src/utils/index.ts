export { isObject, deepMerge } from './deepMerge'
export { default as omit } from './omit'
export * from './mixins'
