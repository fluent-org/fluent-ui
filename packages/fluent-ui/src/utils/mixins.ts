export const findAllByType = (component: any, type: string): any[] => {
  let matches: any[] = []

  if (component.type === type) {
    matches.push(component)
  }

  if (component.children) {
    component.children.forEach(
      (child: any): void => {
        matches = matches.concat(findAllByType(child, type))
      }
    )
  }

  return matches
}
