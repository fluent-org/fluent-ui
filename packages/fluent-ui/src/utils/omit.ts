interface Obj {
  [key: string]: any
}

function omit(obj: Obj, fields: string[]): Omit<Obj, string> {
  const shallowCopy: Obj = {
    ...obj
  }
  for (let i = 0; i < fields.length; i++) {
    const key: string = fields[i]
    delete shallowCopy[key]
  }
  return shallowCopy
}

export default omit
