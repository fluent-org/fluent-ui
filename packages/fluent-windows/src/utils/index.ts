export * from './mixins'
export * from './convertingColor'
