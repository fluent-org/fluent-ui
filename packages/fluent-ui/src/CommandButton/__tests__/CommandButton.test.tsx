import * as React from 'react'
import * as renderer from 'react-test-renderer'
import 'jest-styled-components'

import { ThemeProvider, CommandButton } from '../..'
import { findAllByType } from '../../utils'

describe('CommandButton', (): void => {
  const theme = {}
  test('basic', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <CommandButton>basic</CommandButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('onClick', (): void => {
    const onClick = jest.fn()
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <CommandButton onClick={onClick}>onClick</CommandButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()

    const commandButton = findAllByType(tree, 'button')
    commandButton[0].props.onClick()
    expect(onClick).toHaveBeenCalled()
  })

  test('as', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <CommandButton as="a" href="#">
          as a
        </CommandButton>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
