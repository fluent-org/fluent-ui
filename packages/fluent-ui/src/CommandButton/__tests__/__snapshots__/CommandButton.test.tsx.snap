// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandButton as 1`] = `
.c0 {
  border: none;
  outline: none;
  padding: 10px 22px;
  line-height: 1;
  -webkit-transition: all 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: all 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  background-color: transparent;
}

.c0:hover {
  background-color: rgba(0,0,0,0.1);
}

.c0:active {
  color: #000000;
  background-color: rgba(0,0,0,0.2);
}

<a
  className="c0"
  href="#"
>
  as a
</a>
`;

exports[`CommandButton basic 1`] = `
.c0 {
  border: none;
  outline: none;
  padding: 10px 22px;
  line-height: 1;
  -webkit-transition: all 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: all 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  color: inherit;
  background-color: transparent;
}

.c0:hover {
  background-color: rgba(0,0,0,0.1);
}

.c0:active {
  color: #000000;
  background-color: rgba(0,0,0,0.2);
}

<button
  className="c0"
>
  basic
</button>
`;
